File: app/src/main/java/com/OverSadBoy/samurairise/view/dialog/DialogFragmentConfirm.java
package com.OverSadBoy.samurairise.view.dialog;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.OverSadBoy.samurairise.R;

import java.util.Objects;


public class DialogFragmentConfirm extends DialogFragment {

    private DialogListener listener = null;

    public void setDialogListener(DialogListener listener) {
        this.listener = listener;
    }

    @SuppressLint("InflateParams")
    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext())
                .setView(Objects.requireNonNull(getActivity()).getLayoutInflater().inflate(R.layout.dialog, null))
                .setPositiveButton("Да", (dialog, which) -> listener.positiveClick())
                .setNegativeButton("Нет", (dialog, which) -> listener.negativeClick())
                .create();

    }
}

File: app/src/main/java/com/OverSadBoy/samurairise/view/dialog/DialogFragmentConfirm.kt
package com.OverSadBoy.samurairise.view.dialog

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import androidx.test.runner.AndroidJUnit4
import com.OverSadBoy.samurairise.R
import org.junit.runner.RunWith
import java.util.*

class DialogFragmentConfirm : DialogFragment() {
    private var listener: DialogListener? = null
    fun setDialogListener(listener: DialogListener?) {
        this.listener = listener
    }

    @SuppressLint("InflateParams")
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(context)
                .setView(Objects.requireNonNull(activity).getLayoutInflater().inflate(R.layout.dialog, null))
                .setPositiveButton("Да") { dialog: DialogInterface?, which: Int -> listener.positiveClick() }
                .setNegativeButton("Нет") { dialog: DialogInterface?, which: Int -> listener.negativeClick() }
                .create()
    }
}